src/components/onboardingFormData/PersonalAndBillingDataSection.tsx

```diff
@@ -112,10 +112,16 @@ export default function PersonalAndBillingDataSection({
   const isAooUo = aooSelected || uoSelected;
 
   useEffect(() => {
-    if (isFromIPA && selectedParty?.istatCode) {
-      void getLocationFromIstatCode(selectedParty.istatCode);
+    if (isFromIPA) {
+      if (aooSelected) {
```

Comment:
Written like this it can trigger possible bugs... check the existence of the property not of the entire object